lib/hl7/path.ex

```diff
@@ -0,0 +1,63 @@
+defmodule HL7.Path do
+  @derive Inspect
+
+  defstruct [:data]
+
+  @typep segment :: String.t()
+  @typep indices :: [integer, ...]
+  @type t :: %__MODULE__{
+          data: {segment, indices} | indices
+        }
+
+  @doc """
+  Builds an HL7.Path
+
+  Field paths are used in `HL7.Query` functions to find specific fields in an `HL7.Message`.
+  """
+  @doc since: "0.8.0"
+  @spec new(String.t()) :: t()
+  def new(schema) when is_binary(schema) do
+    use_repeat = String.contains?(schema, "[")
+    use_segment = String.contains?(schema, "-")
+    use_component = String.contains?(schema, ".")
+    chunks = chunk_schema(schema)
+    [head | tail] = chunks
+
+    data =
+      case use_segment do
+        true ->
+          [<<segment::binary-size(3)>> | indices] =
+            case use_component && !use_repeat do
+              false ->
+                [head | tail |> Enum.map(&String.to_integer/1)]
```

Comment:
I think this might read a little bit better with an intermediate var or perhaps just not using a pipe here. wdyt?

Reply:
All of the things in this file were copied in, and I would instead change their contents in something committed to that change alone so I can concentrate on testing it.